lib/execution_engine2/sdk/SDKMethodRunner.py

```diff
@@ -319,6 +317,41 @@ def get_job_status_field(self, job_id, as_admin=False):
         """ Authorization Required: Read """
         return self.get_jobs_status().get_job_status(job_id=job_id, as_admin=as_admin)
 
+    def check_job_batch(
+        self,
+        parent_job_id,
+        check_permission=True,
+        exclude_fields=None,
+        as_admin=False,
+    ):
+        """ Authorization Required: Read """
+
+        if as_admin is True:
+            self.check_as_admin(requested_perm=JobPermissions.READ)
+            check_permission = False
+
+        if exclude_fields and "child_jobs" in exclude_fields:
+            raise ValueError("You can't exclude child jobs from this endpoint")
+
+        parent_job_status = self.get_jobs_status().check_job(
+            job_id=parent_job_id,
+            check_permission=check_permission,
+            exclude_fields=exclude_fields,
+        )
+        child_job_ids = parent_job_status.get("child_jobs")
+        child_job_states = []
+        if child_job_ids:
+            child_job_states = self.get_jobs_status().check_jobs(
+                job_ids=child_job_ids,
+                check_permission=True,
+                exclude_fields=exclude_fields,
+                return_list=1,
+            )["job_states"]
+        return {
+            "parent_jobstate": parent_job_status,
+            "child_jobstates": child_job_states,
```

Comment:
is this going to be an array of all job states, or will it be a mapping with job ID => job state?

Reply:
array of all job states

Reply:
So how are you going to know which job is in which state?

Reply:
@ialarmedalien It is a list of job states, not a list of job statuses. You can check the job[status] field to see the job status.

Reply:
Ah sorry, gotcha!